src/engines/kotor/gui/ingame/partyselection.cpp

```diff
@@ -22,23 +22,145 @@
+		_config.selectedSlots.push_back(_config.forceNPC1);
+		++_numSelectedSlots;
+	}
+
+	if (_config.forceNPC2 != -1) {
+		_config.selectedSlots.push_back(_config.forceNPC2);
+		++_numSelectedSlots;
+	}
+
+	setSlotTexture(0, (_config.trask ? "po_ptrask" : "po_pbastila"));
+
+	for (int i = 0; i < 9; ++i) {
+		toggleSlot(i, _config.slotEnabled[i]);
+
+		bool selected = (_config.forceNPC1 == i) ||
+		                (_config.forceNPC2 == i) ||
+		                (std::find(_config.selectedSlots.begin(),
+		                           _config.selectedSlots.end(),
+		                           i) != _config.selectedSlots.end());
+
+		WidgetCheckBox *checkBox = getCheckBox("BTN_NPC" + Common::composeString(i));
+		checkBox->setState(selected);
+		checkBox->setHighlight(selected);
+	}
+
+	getButton("BTN_BACK")->setDisabled(!config.canCancel);
+
+	onSelectionChanged();
 }
 
 void PartySelectionGUI::callbackActive(Widget &widget) {
 	const Common::UString &tag = widget.getTag();
 	if (tag == "BTN_BACK")
 		_returnCode = 1;
+	else if (tag == "BTN_DONE")
+		_returnCode = 2;
+	else if (tag.beginsWith("BTN_NPC")) {
+		Common::UString tmp(tag);
+		tmp.replaceAll("BTN_NPC", "");
+		int index;
+		Common::parseString(tmp, index);
+		_activeSlot = index;
+		onSelectionChanged();
+	}
+	else if (tag == "BTN_ACCEPT") {
+		if ((_activeSlot != _config.forceNPC1) &&
+		    (_activeSlot != _config.forceNPC2)) {
+			std::vector<int>::iterator i = std::find(_config.selectedSlots.begin(),
+			                                         _config.selectedSlots.end(),
+			                                         _activeSlot);
+			if (i == _config.selectedSlots.end()) {
+				if (_numSelectedSlots < 2) {
+					_config.selectedSlots.push_back(_activeSlot);
+					++_numSelectedSlots;
+				}
+			}
+			else {
+				_config.selectedSlots.erase(i);
+				--_numSelectedSlots;
+			}
+		}
+
+		onSelectionChanged();
+	}
+}
+
+void PartySelectionGUI::toggleSlot(int index, bool enabled) {
+	getLabel("LBL_NA" + Common::composeString(index))->setInvisible(enabled);
+	getLabel("LBL_CHAR" + Common::composeString(index))->setInvisible(enabled);
+}
+
+void PartySelectionGUI::setSlotTexture(int index, const Common::UString &texture) {
+	WidgetCheckBox *checkBox = getCheckBox("BTN_NPC" + Common::composeString(index));
+	checkBox->setFill(texture);
+	checkBox->setSelected(texture);
+	checkBox->setUnselected(texture);
+	checkBox->setSelectedHighlighted(texture);
+	checkBox->setUnselectedHighlighted(texture);
+}
+
+void PartySelectionGUI::onSelectionChanged() {
+	getLabel("LBL_COUNT")->setText(Common::composeString(2 - _numSelectedSlots));
+
+	for (int i = 0; i < 9; ++i) {
+		WidgetCheckBox *checkBox = getCheckBox("BTN_NPC" + Common::composeString(i));
+
+		if (std::find(_config.selectedSlots.begin(),
+		              _config.selectedSlots.end(),
+		              i) != _config.selectedSlots.end()) {
+			checkBox->setBorderColor(0.0f, 1.0f, 0.0f, 1.0f);
+		}
+		else if (_activeSlot == i)
```

Comment:
The else needs to go onto the previous line here as well.

If you think this will look too squished, you can add a new line before the closing brace,